src/metabase/automagic_dashboards/filters.clj

```diff
@@ -0,0 +1,93 @@
+(ns metabase.automagic-dashboards.filters
+  (:require [clojure.string :as str]
+            [metabase.models.field :refer [Field]]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+(def ^:private FieldIdForm
+  [(s/one (s/constrained (s/either s/Str s/Keyword)
+                         (comp #{"field-id" "fk->"} str/lower-case name))
+          "head")
+   s/Any])
+
+(defn- collect-field-references
+  [card]
+  (->> card
+       :dataset_query
+       :query
+       ((juxt :breakout :fields))
+       (tree-seq sequential? identity)
+       (remove (s/checker FieldIdForm))
+       (map last)))
+
+(defn- candidates-for-filtering
+  [cards]
+  (->> cards
+       (mapcat collect-field-references)
+       distinct
+       (map Field)
+       (filter (fn [{:keys [base_type special_type]}]
+                 (or (isa? base_type :type/DateTime)
+                     (isa? special_type :type/Category))))))
+
+(defn- find-fk
+  [from-table to-field]
+  (->> (db/select [Field :id :fk_target_field_id]
+         :fk_target_field_id [:not= nil]
+         :table_id from-table)
+       (filter (comp #{(:table_id to-field)} :table_id Field :fk_target_field_id))
+       (map :id)))
```

Comment:
`u/get-id`

Reply:
I'm intentionally using :id as it will always be an object. Unless we have a convention that we always prefer u/get-id